### DISCUSSION TITLE
Suggest the best way to execute an additional code after confirming user registration

### DISCUSSION THREAD

**Discussion by ademaro** [pallets-eco/flask-security#978] [Q&A] [answered]
After a user has registered and followed the link from the registration confirmation email I want to:
1) add this user to a separate group (add a role)
2) generate invitation codes (essentially another function)

Can you please tell me how to do it better? Is there a standard mechanism or do I have to rewrite the code?

**Comment by jwag956** [comment, marked as answer]
have you looked at:
https://flask-security-too.readthedocs.io/en/stable/api.html#signals

there is a user_confirmed signal that I think would work -
